enhancements/kubelet/kubelet-node-sizing.md

```diff
@@ -0,0 +1,212 @@
+
+### Test Plan
+The following workload can be used to test the automatically generated node sizing values.
+
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: badmem
+spec:
+  replicas: 1
+  selector:
+    app: badmem
+  template:
+    metadata:
+      labels:
+        app: badmem
+    spec:
+      containers:
+      - args:
+        - python
+        - -c
+        - |
+          x = []
+          while True:
+            x.append("x" * 1048576)
+        image: registry.redhat.io/rhel7:latest
+        name: badmem
+
+```
+After submitting this ReplicationController the node should not end up in `NotReady` state. See https://bugzilla.redhat.com/show_bug.cgi?id=1857446 for more information.  
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+    
+  This functionality only modifies the systemd service file of the kubelet. It tries to supply values of `--system-reserved` kubelet flag. As long as kubelet keeps `--system-reserved` flag in place, version skew should not have any impact on this work.  
+    
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+
+  N/A
+
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+  No
+
+## Drawbacks
+
+This solution utilizes kubelet command line flags. Kubelet command line flags have been deprecated in favour of config file, so there is risk for this solution if those flags are actually purged. Having said that, those flags are quite widely used today. So there has not been much traction on actually removing those flags even though they have been marked deprecated. 
+
+## Alternatives
+
+1. Enhance kubelet itself to be more smart about calculating node sizing values. We have an actively debated [KEP](https://github.com/kubernetes/enhancements/pull/2370) in sig-node around this idea. 
+2. Modify MCO the way it handles kubeletconfig. Instead of passing `--system-reserved` argument to the kubelet, maybe there is a possibility to make sure MCO is more tolerant of changes to the kubelet config file. This way we will modify the config file to add system reserve values instead of passing them as `--system-reserved`. 
```

Comment:
This is how I would have expected node sizing to be handled. I would have expected a KubeletConfig resource add by the user to either set explicit values or set it to use auto-sizing.

Reply:
For the day-1 scenario, until the kubelet is deployed and node has joined the cluster the MCO controller does not know the node size.